lib/vm/src/vmoffsets.rs

```diff
@@ -453,13 +457,24 @@ impl VMOffsets {
             .unwrap()
     }
 
+    /// The offset of the trap handler.
+    pub fn vmctx_trap_handler_begin(&self) -> u32 {
+        self.vmctx_builtin_functions_begin()
+            .checked_add(
+                VMBuiltinFunctionIndex::builtin_functions_total_number()
+                    .checked_mul(u32::from(self.pointer_size))
+                    .unwrap(),
+            )
+            .unwrap()
+    }
+
     /// Return the size of the [`VMContext`] allocation.
     ///
     /// [`VMContext`]: crate::vmcontext::VMContext
     pub fn size_of_vmctx(&self) -> u32 {
-        self.vmctx_builtin_functions_begin()
+        self.vmctx_trap_handler_begin()
             .checked_add(
-                VMBuiltinFunctionIndex::builtin_functions_total_number()
+                self.num_trap_handlers
                     .checked_mul(u32::from(self.pointer_size))
                     .unwrap(),
             )
```

Comment:
This seems wrong

Reply:
Why? We just add one more field to vmctx and trap handler is the last field now.